### PR TITLE
[#IC-306] Add Retry for Message View Update failures

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes

- add update message_view function
- add appinsight client
- update io-functions-commons library
- Add `HandleMessageViewUpdateFailures` queue trigger function
- Add utility to handle transient and permanent failures
#### Motivation and Context

Since Event Hub does not support retry logic on function bindings we have to manage retry with another queue if something goes wrong during Message View update.
#### How Has This Been Tested?

Unit test
#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: UpdateCosmosMessageView/function.json [hunk 1]
```diff
@@ -4,10 +4,10 @@
       "type": "eventHubTrigger",
       "name": "rawMessageStatus",
       "direction": "in",
-      "eventHubName": "%MESSAGE_STATUS_TOPIC_NAME%",
-      "consumerGroup": "%MESSAGE_STAUTS_TOPIC_CONSUMER_GROUP%",
+      "eventHubName": "%MESSAGE_STATUS_FOR_VIEW_TOPIC_NAME%",
```
Comment (AleDore) [inline UpdateCosmosMessageView/function.json:7]: typo

File: UpdateCosmosMessageView/function.json [hunk 2]
```diff
@@ -4,10 +4,10 @@
...
+      "consumerGroup": "%MESSAGE_STAUTS_FOR_VIEW_TOPIC_CONSUMER_GROUP%",
```
Comment (AleDore) [inline UpdateCosmosMessageView/function.json:8]: ditto

File: UpdateCosmosMessageView/function.json (hunk 1 above)
Comment (fabriziopapi) [reply UpdateCosmosMessageView/function.json:7]: fixed. thank you!

File: UpdateCosmosMessageView/function.json (hunk 2 above)
Comment (fabriziopapi) [reply UpdateCosmosMessageView/function.json]: fixed.

**Review by fabriziopapi — APPROVED** [review]

File: HandleMessageViewUpdateFailures/handler.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    TE.chain(failureInput =>
+      pipe(
+        failureInput,
+        RetriableHandleMessageViewFailureInput.decode,
+        TE.fromEither,
+        TE.mapLeft(() => toPermanentFailure(Error(failureInput.message))())
+      )
+    ),
+    TE.map(retriableFailure => retriableFailure.body),
+    TE.chain(handleStatusChange(messageViewModel, messageModel, blobService)),
```
Comment (gquadrati) [inline HandleMessageViewUpdateFailures/handler.ts:67]: I prefer to keep the functionality together in the same chain, but I suppose this is very subjective

```suggestion
    TE.chain(failureInput =>
      pipe(
        failureInput,
        RetriableHandleMessageViewFailureInput.decode,
        TE.fromEither,
        TE.mapLeft(() => toPermanentFailure(Error(failureInput.message))()),
        TE.map(retriableFailure => retriableFailure.body)
      )
    ),
    TE.chain(handleStatusChange(messageViewModel, messageModel, blobService)),
```

**Review by gquadrati — APPROVED** [review]
LGTM

**Review by fabriziopapi — APPROVED** [review]

**Review by gquadrati — APPROVED** [review]

**Pull request merged**